### ISSUE TITLE
display-name rule should consider if component is directly assigned to module.exports

### ISSUE THREAD

**Issue by Dashed** [yannickcr/eslint-plugin-react#168]
In babel:
```js
module.exports = React.createClass({
});
```

Output:
```js
"use strict";

module.exports = React.createClass({
  displayName: "exports"
});
```

Demo: https://babeljs.io/repl/#?experimental=true&evaluate=false&loose=true&spec=false&playground=true&code=module.exports%20%3D%20React.createClass(%7B%0A%7D)%3B%0A

-----

babel automatically adds `displayName`; however, this isn't what we want. The display-name rule should warn us to recommend writing code as such:

```js
const Component = React.createClass({
});
module.exports = Component;
```
